From 5ff97b9c1fdf6165417a5d1a1e2aec0588dd9f2f Mon Sep 17 00:00:00 2001
From: Aaron Enye Shi <enye.shi@gmail.com>
Date: Fri, 14 Jun 2024 10:06:13 -0700
Subject: [PATCH] Skip CUPTIRangeProfilerApiTest and CuptiRangeProfilerTest due
 to SEGFAULT in CUDA 12.4 (#951)

Summary:
Pull Request resolved: https://github.com/pytorch/kineto/pull/951

Github Actions CI on NVIDIA A10G instances are failing for these tests since the upgrade to 12.4:

19 - CuptiRangeProfilerApiTest.asyncLaunchUserRange (SEGFAULT)
20 - CuptiRangeProfilerApiTest.asyncLaunchAutoRange (SEGFAULT)
24 - CuptiRangeProfilerTest.UserRangeTest (SEGFAULT)
25 - CuptiRangeProfilerTest.AutoRangeTest (SEGFAULT)

We are tracking the issue here: https://github.com/pytorch/kineto/issues/949

Test Plan: CI

Reviewed By: sraikund16

Differential Revision: D58588836

Pulled By: aaronenyeshi

fbshipit-source-id: 4b1c02d18e235d1c8a4f7c5162d59950cfa89dcb
---
 libkineto/test/CMakeLists.txt | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/libkineto/test/CMakeLists.txt b/libkineto/test/CMakeLists.txt
index efb35fa..2c2a487 100644
--- a/libkineto/test/CMakeLists.txt
+++ b/libkineto/test/CMakeLists.txt
@@ -65,7 +65,9 @@ target_include_directories(CuptiRangeProfilerApiTest PRIVATE
     "${LIBKINETO_DIR}/src"
     "${CUDA_SOURCE_DIR}/include"
     "${CUPTI_INCLUDE_DIR}")
-gtest_discover_tests(CuptiRangeProfilerApiTest)
+# Skipping due to SEGFault in 12.4
+# Tracked here: https://github.com/pytorch/kineto/issues/949
+# gtest_discover_tests(CuptiRangeProfilerApiTest)
 
 # CuptiRangeProfilerConfigTest
 add_executable(CuptiRangeProfilerConfigTest CuptiRangeProfilerConfigTest.cpp)
@@ -94,7 +96,9 @@ target_include_directories(CuptiRangeProfilerTest PRIVATE
     "${LIBKINETO_DIR}/src"
     "${CUDA_SOURCE_DIR}/include"
     "${CUPTI_INCLUDE_DIR}")
-gtest_discover_tests(CuptiRangeProfilerTest)
+# Skipping due to SEGFault in 12.4
+# Tracked here: https://github.com/pytorch/kineto/issues/949
+# gtest_discover_tests(CuptiRangeProfilerTest)
 
 # CuptiStringsTest
 add_executable(CuptiStringsTest CuptiStringsTest.cpp)
-- 
2.44.2

